pkg/bundle/read.go

```diff
@@ -153,9 +154,12 @@ func read(ctx context.Context, name, baseDir string, bundleSpecReader io.Reader,
 
 	def.Spec.ForceSyncGeneration = opts.SyncGeneration
 
-	def, err = appendTargets(def, opts.TargetsFile)
-	if err != nil {
-		return nil, err
+	// If override targets is set do not append other targets
+	if !bundle.OverrideTargets && len(def.Spec.Targets) > 0 {
+		def, err = appendTargets(def, opts.TargetsFile)
```

Comment:
What happens if `OverrrideTargets` is `false`, but the length is `<= 0`?

Reply:
yeah, i think i was trying to be too fancy. I think if `OverrrideTargets` is `false` append targets. Think the scenario i was worried about was if `OverrrideTargets` is `true` but no targets were defined, we would want to append defaults.

Reply:
I actually think that's fine! I was asking whether or not that scenario is one we should account for still.

Reply:
I'm not sure. Part of me says that `override` should be authoritative and if you end up with 0 targets, then it should be removed. But a safety net is good as well.

I think my original logic was wrong, i think this would be the `if` statement to handle the safety net scenario.
```go
 	if !bundle.OverrideTargets || (bundle.OverrideTargets && len(def.Spec.Targets) > 0) { }
```

Reply:
Think my last commit resolves this.